src/python/pants/backend/project_info/peek.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import collections.abc
+import json
+import os
+from dataclasses import asdict, is_dataclass
+from enum import Enum
+from typing import Iterable, cast
+
+from pkg_resources import Requirement
+
+from pants.engine.addresses import Address, Addresses, BuildFileAddress
+from pants.engine.console import Console
+from pants.engine.fs import DigestContents, FileContent, PathGlobs
+from pants.engine.goal import Goal, GoalSubsystem, Outputting
+from pants.engine.rules import Get, MultiGet, collect_rules, goal_rule
+from pants.engine.target import Target, UnexpandedTargets
+
+
+class OutputOptions(Enum):
+    RAW = "raw"
+    JSON = "json"
+
+
+class PeekSubsystem(Outputting, GoalSubsystem):
+    """Display BUILD file info to the console.
+
+    In its most basic form, `peek` just prints the contents of a BUILD file. It can also display
+    multiple BUILD files, or render normalized target metadata as JSON for consumption by other
+    programs.
+    """
+
+    name = "peek"
+
+    @classmethod
+    def register_options(cls, register):
+        super().register_options(register)
+        register(
+            "--output",
+            type=OutputOptions,
+            default=OutputOptions.JSON,
+            help="which output style peek should use",
+        )
+        register(
+            "--exclude-defaults",
+            type=bool,
+            default=False,
+            help="whether to exclude values matching default field values from normalized output",
```

Comment:
Ditto on capitalizing. 

With docs, imo, it can sometimes be helpful to say the same thing in a different way. My thinking is if the person is not familiar with that word, such as English language learners. Rather than repeating "exclude", you can use something like "Whether to leave off values that match Pants's default values when using `json` output."